src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/AtlasDiffCommand.java

```diff
@@ -272,10 +247,165 @@ public void registerOptionsAndArguments()
                 GEOJSON_CONTEXT);
         registerOption(FULL_OPTION_LONG, FULL_OPTION_DESCRIPTION, OptionOptionality.REQUIRED,
                 FULL_CONTEXT);
+        registerOption(FOLDER_SEARCH_RECURSIVE_OPTION_LONG,
+                FOLDER_SEARCH_RECURSIVE_OPTION_DESCRIPTION, OptionOptionality.OPTIONAL,
+                FULL_CONTEXT, GEOJSON_CONTEXT, LDGEOJSON_CONTEXT);
         registerArgument(BEFORE_ATLAS_ARGUMENT, ArgumentArity.UNARY, ArgumentOptionality.REQUIRED,
                 DEFAULT_CONTEXT, LDGEOJSON_CONTEXT, GEOJSON_CONTEXT, FULL_CONTEXT);
         registerArgument(AFTER_ATLAS_ARGUMENT, ArgumentArity.UNARY, ArgumentOptionality.REQUIRED,
                 DEFAULT_CONTEXT, LDGEOJSON_CONTEXT, GEOJSON_CONTEXT, FULL_CONTEXT);
         super.registerOptionsAndArguments();
     }
+
+    boolean compute(final AtlasDiffCommandContext context, final File beforeAtlasFile,
+            final File afterAtlasFile)
+    {
+        boolean result = false;
+        final Map<String, File> beforeNamesToFiles = new HashMap<>();
+        final Map<String, File> afterNamesToFiles = new HashMap<>();
+        final List<File> beforeFilesToConsider = context.isRecursive()
+                ? beforeAtlasFile.listFilesRecursively(false)
+                : beforeAtlasFile.listFiles(false);
+        final List<File> afterFilesToConsider = context.isRecursive()
+                ? afterAtlasFile.listFilesRecursively(false)
+                : afterAtlasFile.listFiles(false);
+        beforeFilesToConsider.stream().filter(FileSuffix.ATLAS::matches).forEach(
+                file -> beforeNamesToFiles.put(getRelativeFileName(beforeAtlasFile, file), file));
+        afterFilesToConsider.stream().filter(FileSuffix.ATLAS::matches).forEach(
+                file -> afterNamesToFiles.put(getRelativeFileName(afterAtlasFile, file), file));
```

Comment:
If you remove the `FileSuffix` filter here, then this could support any existing (or future) atlas format (e.g. text atlases with `.txt`/`.atlas.txt` suffix). My below note for how to handle load errors.

Reply:
I see what you mean here, however I'd need to find a way to detect what is a valid atlas at this point, and not below, since I need to also list the Atlas files that are in one folder and not in another.
Maybe I need to add a function in AtlasResourceLoader which tells me what is a recognized atlas format?

Reply:
Interesting ok. Let me take a closer look at this and get back to you here. There might be a way to easily do this, and then you could include it in the PR!

Reply:
So there is a way to do this! `AtlasResourceLoader` declares a couple public predicates: `HAS_ATLAS_EXTENSION` and `HAS_TEXT_ATLAS_EXTENSION`. It also has a predicate `CONTENTS_LOOK_LIKE_TEXT_ATLAS`. You could have a `checkIsAtlas` method that takes a resource and returns something like `(resource.hasTextAtlasExtension || resource.hasAtlasExtension) || resource.contentsLookLikeTextAtlas`. That would catch things like `file.atlas`, `file.atlas.txt`, and even things like `my_atlas.txt` in the case that is a text atlas but the person used a different extension.

Then in the future if `AtlasResourceLoader` ever gets new file type capabilities, it would be easy to just update this `checkIsAtlas` method.